src/d3a/d3a_core/sim_results/bills.py

```diff
@@ -232,8 +232,7 @@ def _update_market_fees(self, area_dict, area_core_stats):
 
     def _accumulate_grid_fee_charged(self, area_dict, area_core_stats):
         area_stats = area_core_stats.get(area_dict['uuid'])
-        for trade in area_stats.get('trades', []):
-            self.cumulative_fee_charged_per_market += trade['fee_price'] / 100.
+        self.cumulative_fee_charged_per_market += area_stats.get('market_fee', 0.) / 100.
```

Comment:
Ahh, so you have access to the market_fee at the end. Great! So my suggestion was not that dumb! Thanks!

Reply:
I still wonder why the former implementation failed though. Nevermind.

Reply:
@spyrostz it counted the same fee several times, Faizan suggested because of chain trading, when it goes up and then down in hierarchy